process/smartContract/process.go

```diff
@@ -300,26 +301,42 @@ func (sc *scProcessor) createVMCallInput(tx *transaction.Transaction) (*vmcommon
 	return vmCallInput, nil
 }
 
-func (sc *scProcessor) createVMDeployInput(tx *transaction.Transaction) (*vmcommon.ContractCreateInput, error) {
+func (sc *scProcessor) createVMDeployInput(
+	tx *transaction.Transaction,
+) (*vmcommon.ContractCreateInput, []byte, error) {
 	vmInput, err := sc.createVMInput(tx)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	if len(vmInput.Arguments) < 1 {
+		return nil, nil, process.ErrNotEnoughArgumentsToDeploy
+	}
+	// first argument after the code is the vmType
+	vmAppendedType := make([]byte, endVMTypeIndex-startVMTypeIndex)
+	vmType := vmInput.Arguments[0].Bytes()
+	vmArgLen := len(vmType)
+	if vmArgLen > endVMTypeIndex-startVMTypeIndex {
+		return nil, nil, process.ErrVMTypeLengthInvalid
 	}
 
+	copy(vmAppendedType[endVMTypeIndex-startVMTypeIndex-vmArgLen:], vmType)
```

Comment:
Might extract this in a func having a descriptive name like: applyVmTypesBytes()

Reply:
done